### DISCUSSION TITLE
Online indicator

### DISCUSSION THREAD

**Discussion by huskylogic** [amidaware/tacticalrmm#1371] [Ideas]
I love the bars that tell when the last time the agent communicated, but I'd love to see an indicator on an indicator on the main screen to tell when the agent is disconnected meaning the computer is turned off.

**Comment by dinger1986** [comment]
can you mock up an example of what you mean? At the moment the bars are Red when Overdue, Yellow when Offline and Green when Connected.

**Comment by huskylogic** [comment]
An example of another RMM that does this is pictured below.  When the unit is turned on it says “Connected”. No matter if that unit was turned on last week or just five minutes ago.  It indicates to me that I can click and take control of this unit.  That never changes until the system is turned off.  Right now the TRMM indicators can show red even though the unit is turned on.  For example the agent that shows “Connected” in my image below, shows red on the TRMM version.  

 

When I change the time frame for the color coded system, it will still show green as “Connected” even when the unit is turned off. 

 



 

Fred Claus
Technical Director

Husky Logic
p: 716-775-7002

e: ***@***.***
w: HuskyLogic.com

**Comment by dinger1986** [comment]
sorry I dont see a picture, maybe best to reply to the thread rather than email

**Comment by huskylogic** [comment]
![image](https://user-images.githubusercontent.com/73186760/206755131-b7bf0e1c-8faf-401c-89c8-ff7faaddb9ad.png)
The attached picture is the sample I was trying to show you.  The column that says "Connected" or "Disconnected" is the one we are looking at.  As long as this computer is on and connected to the internet it shows "Connected".  that same computer on the TRMM side shows red, but I am still able to do things on it.

**Comment by dinger1986** [comment]
ah ok, so when its showing red are you refreshing the screen? Its maybe that its not refreshing fast enough

**Reply by huskylogic** [reply]
Yes, and I just refreshed again just now. I'm still seeing red indicators. The top one in this image is the one that shows connected on the other one. 
![image](https://user-images.githubusercontent.com/73186760/206756910-95f6d8a5-b60d-432b-bda4-e495f577111c.png)

**Reply by dinger1986** [reply]
Ok makes sense.

So if you "take control" does it show up connected? Is the service running etc?

**Comment by huskylogic** [comment]
Sorry just saw this one. Yes I refresh the screen and I still get this red indicator. But I am still able to connect to that device if it is actually online even if the red indicator is showing.

**Comment by silversword411** [comment]
Hopefully this will help with questions on what's all the stuff in the TRMM, how and when it's updating and refreshing. 

Note this is the TRMM agent, and not related to mesh. 

![image](https://user-images.githubusercontent.com/13395348/207724179-633fb4ed-f716-47f5-a16c-8eb5bb486382.png)

**Reply by huskylogic** [reply]
This I think clears up a lot for me. The TRMM agent is the one responsible for the red icons.  That is just the management agent communicating with the server.  The Mesh agent is the one that tells me if I can connect to the computer or not.  Correct?  I wonder if it would be possible to have an icon where we can see the status of the mesh agent if that's the case.  I think I understand it now. Thanks for sharing this document.

**Reply by silversword411** [reply]
Correct. I usually use connect buttons appearing in the mesh-related dialogs (or goto mesh.yourdomain.com and look at the online status)

Use the repair/recover buttons to try and get mesh back if it's not working.

**Comment by huskylogic** [comment]
By the way I tried to refresh the indicator by clicking the icon shown in the previous message and it doesn't work properly for me.  I know the agent is online and in most cases (I didn't check every one of them) I can take control of the computer. However the "remote background" option shows me this message when I try to access it on an agent with the red or yellow indicators. 
![image](https://user-images.githubusercontent.com/73186760/207887941-ab7dc6ae-7d14-4062-b74a-5b837bdefd4d.png)

**Reply by silversword411** [reply]
You're either running #unsupported proxy or have a problem with server and running https://docs.tacticalrmm.com/troubleshooting/#server-troubleshooting-script will point you to the problem
